pep/003-dependency-namespaces-and-labels.md

```diff
@@ -0,0 +1,508 @@
+Since we are pre 1.0, it's an acceptable breaking change but we will need to warn people before the release and call it out in the release notes.
+
+___
+
+## Security Implications
+
+* Reusing outputs from existing installations
+
+  This proposal allows you to reuse outputs from an existing installation.
+  Porter does not have any ACLs on installation data, you either have access to the host environment or you don't.
+  The way to isolate installation data from being accessed or reused by another installation is to use separate host environments.
+  Users could do this previously through `porter installation outputs show OUTPUT --installation INSTALLATION` so it's not a new concern, just makes it easier.
+
+___
+
+## Rejected Ideas
+
+* We considered the seemingly obvious solution of scanning a OCI registry for bundles that implement an interface, and rejected it because:
+  * Use Case: The use case isn't to not have to think about which bundle is used for a dependency but being able to select from a variety of implementations.
+  * Fit for Purpose: The implementation rarely doesn't matter and relies on the target environments infrastructure. Selecting an implementation and bundle author at random wouldn't make anyone happy.
+  * Performance: Scanning registries would significantly slow down bundle installation.
+  * Predictibility: A different bundle could be selected depending on the search order and the contents of the registry on any day.
+
+
+---
+
+## Open Questions
+
+* I'm not sure yet how to handle passing credentials to the dependency that implements an interface, as that isn't possible to standardize in the interface.
+  An azure mysql bundle will require different credentials than the google mysql bundle.
+* Should the installation name label selector be defined in porter.yaml (i.e. a hard coded or parameterized dependency name) or should it only be a CLI flag?
+  * I lean towards just a flag because we don't want people hard coding dependencies based on a dynamic value (installation name) or accidentally encourage people to make parameters to set dependency names when there are flags to do that.
+
+* Are we okay with directly exposing the mechanisms for namespace or do we think there's value in making them more task based?
+  I'm not sure that I have a use case for allowing namespace to be set to anything, we certainly don't want it hardcoded or again invite people to create parameters for it.
+  For example instead of `namespace: "{{ installation.namespace }}"`, we could use `scope: namespace|global` which translates to that under the covers.
+  By limiting the values in the porter.yaml, we would be making it easier to do it the right way.
+* For the `dependencies.lifecycle` do we need both managed and the actions? I'm concerned it will confuse people since they are intended to be one or the other and not merged.
```

Comment:
Hmmm good Q... not sure here myself.  Good to raise, though.

Reply:
Deferring to the follow-up lifecycle management proposal.